### PR TITLE
Fusion Filtering Staggered PR1

### PR DESCRIPTION
**Pull request by runjin326**
### Purpose/implementation Section
Add R script for generating reference normal expression matrix

#### What was your approach?
 
Use histologies file to select corresponding columns in the expression matrix
Update run script to capture file name changes and include run script to generate reference file

#### What GitHub issue does your pull request address?
https://github.com/PediatricOpenTargets/ticket-tracker/issues/7

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes.

#### What types of results are included (e.g., table, figure)?
Filtered expression matrix of only normal GTEx samples of adrenal gland 
Altered run script for running all the R scripts on this modified module

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Review by kgaonkar6 — CHANGES_REQUESTED** [review]
Thanks @runjin326! Just 1 change required before I can approve.

Do you also want to add an update to the README mentioning this new script or will you have that updated in a downstream PR?

File: analyses/fusion_filtering/run_fusion_merged.sh [hunk 1]
```diff
@@ -27,39 +27,43 @@ results_path="results/"
...
 # data release files to use for recurrent fusion/fused genes detection
 
-putative_oncogenic_fusion="${results_path}/pbta-fusion-putative-oncogenic.tsv"
+putative_oncogenic_fusion="${results_path}/fusion-putative-oncogenic.tsv"
 
+# Run filtering code to get the reference file
+Rscript 00-normal-matrix-generation.R  --expressionMatrix $rna_expression_file \
+                                       --clinicalFile $histologies_file \
+                                       --specimenType "Adrenal Gland"
+                                       --outputfile "${references_path}/gtex_adrenal_gland_TPM_hg38.rds"
```
Comment (kgaonkar6) [inline analyses/fusion_filtering/run_fusion_merged.sh]:
```suggestion
# Run filtering code to get the reference file
Rscript 00-normal-matrix-generation.R  --expressionMatrix $rna_expression_file \
                                       --clinicalFile $histologies_file \
                                       --specimenType "Adrenal Gland" \
                                       --outputfile "${references_path}/gtex_adrenal_gland_TPM_hg38.rds"
```

**Comment by kgaonkar6** [comment]
`references/gtex_adrenal_gland_TPM_hg38.rds` is missing in this PR, could you rerun the code and add it here so we have the output from the script?

**Comment by runjin326** [comment]
Thanks @kgaonkar6 ! I added the back slash and include the reference file.

File: analyses/fusion_filtering/run_fusion_merged.sh (hunk 1 above)
Comment (runjin326) [reply analyses/fusion_filtering/run_fusion_merged.sh]: I will also modify this for all the following PRs. The README is included in PR7 but currently it hasn't mentioned this particular script - I will modify that. Thanks!

**Comment by kgaonkar6** [comment]
Thanks for the update @runjin326! 

Looking at closly as PediatricOpenTargets/ticket-tracker#7. 

I see the comment
>for neuroblastoma, we will have to generate this file for the GTEX adrenal gland 

The cohort GMKF is neuroblastoma , which commonly occur in adrenal glands so for GMKF we will use the `reference/gtex_adrenal_gland_TPM_hg38.rds` as the normal_expression_file. 

But as you know we are also using PBTA cohort samples, which are brain tumors. So we need to create a normal_expression_file for gtex Brain group as well. And then use each normal expression file for the zscoring.

Let me know if you have any questions.

**Comment by runjin326** [comment]
@kgaonkar6, thanks for the clarification! So that means we need to zscore correct GMKF with adrenal gland and then zscore correct PBTA with brain and then `rbind` those to a final table (in script 03), correct? Also, for the `filter(cohort == "PBTA" | cohort == "GMKF") ` suggested changes, did you mean to make cohort as an input arg as well?

**Comment by kgaonkar6** [comment]
I think we can make `cohort` a param in the function `ZscoredAnnotation`, since each cohort would need to use a different normal_expression_file, like PBTA would use a normal expression file which filters with `gtex_group=="Brain"` and GMKF will use `gtex_group=="Adrenal Gland"`
 And then as you said rbind and as final output from `03`

**Comment by runjin326** [comment]
@kgaonkar6, now Rscript 00 are ran twice to generate both the brain and adrenal gland - however, I did not push the brain matrix since it's over 200M.

**Review by kgaonkar6 — APPROVED** [review]
LGTM! 
```
> check_adrenal <- readRDS("references/gtex_adrenal_gland_TPM_hg38.rds")
> dim(check_adrenal)
[1] 36718   258
> check_brain <- readRDS("references/gtex_brain_TPM_hg38.rds")
> dim(check_brain)
[1] 36718  2642
```

File: analyses/fusion_filtering/run_fusion_merged.sh [hunk 2]
```diff
@@ -27,39 +27,48 @@ results_path="results/"
...
 
 
 # relevant gene expression files
-polya_expression_file="${data_path}/pbta-gene-expression-rsem-fpkm.polya.rds"
-stranded_expression_file="${data_path}/pbta-gene-expression-rsem-fpkm.stranded.rds"
+rna_expression_file="${data_path}/gene-expression-rsem-tpm-collapsed.rds"
 
 # reference expression file
-normal_expression_file="${references_path}/Brain_FPKM_hg38_matrix.txt.zip"
+normal_expression_file="${references_path}/gtex_adrenal_gland_TPM_hg38.rds"
```
Comment (kgaonkar6) [inline analyses/fusion_filtering/run_fusion_merged.sh]: do you want to remove this since you have the name hardcoded in the `00` script?

Comment (runjin326) [reply analyses/fusion_filtering/run_fusion_merged.sh]: Good call - I modified it to be not hardcoded since this will be used in 03 script as well.

File: analyses/fusion_filtering/run_fusion_merged.sh [hunk 3]
```diff
@@ -27,39 +27,49 @@ results_path="results/"
...
-arriba_file="${data_path}/pbta-fusion-arriba.tsv.gz"
-starfusion_file="${data_path}/pbta-fusion-starfusion.tsv.gz"
+arriba_file="${data_path}/fusion-arriba.tsv.gz"
+starfusion_file="${data_path}/fusion-starfusion.tsv.gz"
 standard_arriba_file="${scratch_path}/arriba.tsv"
 standard_starfusion_file="${scratch_path}/starfusion.tsv"
 
 # general filtering parameters
-artifact_filter="GTEx_Recurrent|DGD_PARALOGS|Normal|BodyMap"
+artifact_filter="GTEx|HGNC_GENEFAM|DGD_PARALOGS|Normal|BodyMap"
```
Comment (jharenza) [inline analyses/fusion_filtering/run_fusion_merged.sh]: I thought I saw @kgaonkar6 also comment on this, but I am not sure what the downstreaam implications of this change are - why is this changed?

Comment (runjin326) [reply analyses/fusion_filtering/run_fusion_merged.sh]: Yes. So if I leave the GTEx_recurrent there, it would throw an error message saying that "GTEx_Recurrent" does not exist or something - this filter used here is based on the sample script of 02. But I can use the original setup if you think this is more accurate.

Comment (jharenza) [reply analyses/fusion_filtering/run_fusion_merged.sh]: @kgaonkar6 has there been some update in FusionAnnotator that dropped this?

Also concerned about `HGNC_GENEFAM` - what does adding this do?

Comment (jharenza) [reply analyses/fusion_filtering/run_fusion_merged.sh]: Per our [annoFuse paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03922-7), 

```Of note, we decided not to remove genes annotated in HGNC_GENEFAM, as this database contains multiple oncogenes and their removal resulted in missed true fusions using our validation truth set. ```

So, I do not think we want to add this filter now.

Comment (kgaonkar6) [reply analyses/fusion_filtering/run_fusion_merged.sh]: There has been no update in the FusionAnnotator used, we just don't have the `GTEx_Recurrent` and `GTEx` as you will see from searching `GTEx_Reccurrent` in page here: https://github.com/FusionAnnotator/FusionAnnotator/wiki/Home/3953b06bebecb0703acb823ae562c8f11fe69ce4

So only if it has `GTEx_Reccurrent` then we remove it.

Since the PR is only related to script `00` this specific issue is not discussed here even though bash script was updated  for the overall script from the original PR which had all the updates.

Comment (kgaonkar6) [reply analyses/fusion_filtering/run_fusion_merged.sh]: This change is discussed part of the PR 2 here https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/37#discussion_r664585166

Comment (jharenza) [reply analyses/fusion_filtering/run_fusion_merged.sh]:
```suggestion
artifact_filter="GTEx|DGD_PARALOGS|Normal|BodyMap"
```

Comment (kgaonkar6) [reply analyses/fusion_filtering/run_fusion_merged.sh]: Hi @runjin326  sorry @jharenza discussed it further, there are some fusion annotated as `GTEx` that are also seen in cancer databases so we don't want to remove them here. 

 Can you keep the artifact filter as it was from the original script. That we are consistent with previous projects `GTEx_Recurrent|DGD_PARALOGS|Normal|BodyMap`.


Also, just confirming, you get this warning not an error right?
```
Warning message:
In fusion_filtering_QC(standardFusioncalls = standardFusioncalls,  :
  No fusion calls with annotation: GTEx_Recurrent
```

Comment (runjin326) [reply analyses/fusion_filtering/run_fusion_merged.sh]: Correct - this is what I got - just to confirm, we keep it as GTEx_Recurrent not GTEx and it's ok to get the warning correct?

Comment (kgaonkar6) [reply analyses/fusion_filtering/run_fusion_merged.sh]: Correct, we have the warning coded to just let the user know that we didn't find `fusion calls with annotation: GTEx_Recurrent`.

File: analyses/fusion_filtering/run_fusion_merged.sh (hunk 1 above)
Comment (kgaonkar6) [reply analyses/fusion_filtering/run_fusion_merged.sh]: Could you add the 00 script description here? 

Basically, by checking off the checkbox below we want to check ourselves and also let reviewer know that the chnages in each PR having required README updated 

- [ ] This analysis module has a README and it is up to date.

**Review by kgaonkar6 — APPROVED** [review]
README is updated in #42  to add description for `00`.

**Comment by kgaonkar6** [comment]
Perfect thanks for update @runjin326! Looks ready for a merge now.
